tests/test_cli_patterns_rag.py

```
@@ -0,0 +1,183 @@
+            }
+
+        def apply_tuning(self, project_root, recs):  # noqa: ARG002
+            return SimpleNamespace(project_path=str(project_root), project_id="proj-1")
+
+    monkeypatch.setattr("refactron.cli.patterns.RuleTuner", _FakeTuner)
+
+    class _FakeProfile:
+        project_id = "proj-1"
+        project_path = "x/y"
+        last_updated = datetime.now(timezone.utc)
+        enabled_patterns = {"p1"}
```

Ruff (0.15.6)

[warning] Mutable default value for class attribute

(RUF012)